File: Fce.php
<?php
namespace langzi;
error_reporting(E_ALL);
require_once __DIR__ . '/Thrift/ClassLoader/ThriftClassLoader.php';

use Thrift\ClassLoader\ThriftClassLoader;

$GEN_DIR = realpath(__DIR__.'/gen-php');

$loader = new ThriftClassLoader();
$loader->registerNamespace('Thrift', __DIR__);
$loader->registerDefinition('langzi', $GEN_DIR);
$loader->register();

use Thrift\Protocol\TBinaryProtocol;
use Thrift\Protocol\TMultiplexedProtocol;
use Thrift\Transport\THttpClient;
use Thrift\Transport\TBufferedTransport;
use Thrift\Exception\TException;

try {
$socket = new THttpClient('localhost', 9090, '/Fse.php');
//   $socket = new THttpClient('localhost', 8090, '/thrift/penngo/PhpMulServer.php');
  $transport = new TBufferedTransport($socket);
  $protocol = new TBinaryProtocol($transport);

  
  $upNameProtocol = new TMultiplexedProtocol($protocol, 'upName');
  $upName = new upNameClient($upNameProtocol);
  $member = $upName->EditName(12, 'zl');
  var_dump($member);

  $upAddressProtocol = new TMultiplexedProtocol($protocol, 'upAddress');
  $upAddress = new upAddressClient($upAddressProtocol);
  $member = $upAddress->EditAddress(12, 'ld');
  var_dump($member);

  $upPwdProtocol = new TMultiplexedProtocol($protocol, 'upPwd');
  $upPwd = new upPwdClient($upPwdProtocol);
  $member = $upPwd->EditPassword(12, 'ld123');
  var_dump($member);
//   $transport->close();
} catch (TException $tx) {
  print 'TException: '.$tx->getMessage()."\n";
}

File: py/Fce.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-


import sys
from thrift.protocol.TMultiplexedProtocol import TMultiplexedProtocol
from thrift import Thrift
from thrift.transport import TSocket
from thrift.transport import TTransport
from thrift.protocol import TBinaryProtocol
sys.path.append('./../gen-py')


from first import upAddress
from first import upName
from first import upPwd


# TMultiplexProtocol

transport = TSocket.TSocket('localhost', 9090)
transport = TTransport.TBufferedTransport(transport)
protocol = TBinaryProtocol.TBinaryProtocol(transport)

transport.open()
# 如果服务端使用TMultiplexedProcessor接收处理，客户端必须用TMultiplexedProtocol并且指定serviceName和服务端的一致
upname = TMultiplexedProtocol(protocol, "upname")
upaddress = TMultiplexedProtocol(protocol, "upaddress")
uppwd = TMultiplexedProtocol(protocol, "uppwd")

upname_client = upName.Client(upname)  # msg客户端
upaddress_client = upAddress.Client(upaddress)  # user客户端
uppwd_client = upPwd.Client(uppwd)  # user客户端


print(upname_client.EditName(1, 'nzl123'))
print(upaddress_client.EditAddress(1, 'address'))
print(uppwd_client.EditPassword(1, 'password'))

transport.close()


# try:
#     transport = TSocket.TSocket('localhost', 9090)
#     transport = TTransport.TBufferedTransport(transport)
#     protocol = TBinaryProtocol.TBinaryProtocol(transport)

#     transport.open()
#     upNameclient = upName.Client(protocol)

#     print("client - EditName")
#     print("server - " + upNameclient.EditName(2, 'name'))

#     client = upAddress.Client(protocol)
#     print("client - EditAddress")
#     print("server - " + client.EditAddress(2, 'address'))

#     # print("client - say")
#     # msg = client.say("Hello!")
#     # print("server - " + msg)

#     transport.close()

# except Thrift.TException as ex:
#     print("%s" % (ex.message))
